be/src/olap/rowset/segment_v2/page_io.cpp

```diff
@@ -112,7 +112,7 @@ Status PageIO::read_and_decompress_page(const PageReadOptions& opts, PageHandle*
     auto cache = StoragePageCache::instance();
     PageCacheHandle cache_handle;
     StoragePageCache::CacheKey cache_key(opts.rblock->path(), opts.page_pointer.offset);
-    if (opts.use_page_cache && cache->lookup(cache_key, &cache_handle)) {
+    if (opts.use_page_cache && cache->is_cache_available(opts.type) && cache->lookup(cache_key, &cache_handle, opts.type)) {
```

Comment:
I think the check of `is_cache_available()` can be put into `lookup()` method for easy to use?
As well as the following `insert` logic

Reply:
I choose to make the `is_cache_available()` check as an interface basically for two reasons:

1. If the check is failed (which means that the percentage is set to 0 or 100), the `lookup()` and `insert()` method will not be called, which could save some of the cost of parameter passing.
2. It will do minimal changes to the original `read_and_decompress_page()` method in page_io.cpp and remain its design on whether call the `lookup()` and `insert()` methods.